### ISSUE TITLE
Zero RPN bias 

### ISSUE THREAD

**Issue by LowBoonKeong** [ruotianluo/pytorch-faster-rcnn#53]
Hi @ruotianluo , i did a train on new model on PASVAL_VOC dataset with Vgg16 (CPU only) downloaded from https://github.com/jcjohnson/pytorch-vgg. However, i found that there is always zero at the bias of rpn_net, rpn_cls_score_net, rpn_bbox_pred_net, cls_score_net, bbox_pred_net. Is it an error ? If yes, may i know which part should be modified or change ?

**Comment by ruotianluo** [comment]
It looks like an error.  The biases are initialized to be 0, but they should be updated. Can you check if the learning rate of biases are greater than 0 and the weight decay of biases is not too large.

**Comment by LowBoonKeong** [comment]
Hi @ruotianluo , can you show me the line of learning rate and weight decay of biases ? I changed the following few lines in "network.py", but still get the same error. 
![13](https://user-images.githubusercontent.com/30174905/34324350-27f5f63c-e8ab-11e7-94f5-e92f7dad419b.PNG)
to 
![image](https://user-images.githubusercontent.com/30174905/34324354-4ed7d64e-e8ab-11e7-953d-ac5c1f65c66b.png)
My train.learning_rate=0.0001 and train.weight_decay=0.0001

**Comment by ruotianluo** [comment]
https://github.com/ruotianluo/pytorch-faster-rcnn/blob/master/lib/model/train_val.py#L126
Here is the place where I set the learning rate and weight decay.

**Comment by LowBoonKeong** [comment]
Hi @ruotianluo , thank you very much. The problem solved after setting the flag TRAIN.BIAS_DECAY.

**Comment by LowBoonKeong** [comment]
Hi @ruotianluo , I would like to be appreciate if you could provide some idea or solution for me about the inaccurate localization box shown on figure below. 
![image](https://user-images.githubusercontent.com/30174905/34827418-0daa436e-f715-11e7-9954-6a3d69a663a7.png)
My learning rate = weight decay=0.0001, crop pooling, rpn = 7x7
Thank you

**Comment by ruotianluo** [comment]
Did you change the anchor sizes you use?

**Comment by LowBoonKeong** [comment]
yes, I use anchor scale = (4, 8, 16) and anchor ratio = (0.5, 1, 2). In generate_anchors.py, the base_size = 16, should the base_size change to some smaller value ?

**Comment by ruotianluo** [comment]
You may need to change the ratios to some more suitable values because most texts are lower in height.

The result is still wierd, but the good side is the left up corner is correct. So it's definitely doing something correctly. Maybe it's because you prepared the data incorrectly? You shifted the height and width??

I'm not sure.

**Comment by LowBoonKeong** [comment]
Hi @ruotianluo , thank you very much. The error is solved after reverse width and height.
